Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -421,4 +423,71 @@ public Apis.Bigquery.v2.Data.Job DoExtract()
         }
     }
 
+    /// <summary>
+    /// <para type="synopsis">
+    /// Returns the result of a completed BQ Job.
+    /// </para>
+    /// <para type="description">
+    /// Returns the result of a completed BQ Job. Requires the ‘Reader’ dataset role. You can specify 
+    /// how long the call should wait for the query to be completed, if it is not already finished. 
+    /// This is done with the -Timeout parameter. An integer number of seconds is taken, and the 
+    /// default is 10. This cmdlet returns BigQueryRow objects.
+    /// </para>
+    /// <example>
+    ///   <code>
+    /// PS C:\> $job = Start-BqJob -Query "select * from book_data.classics"
+    /// PS C:\> $job | Receive-BqJob -Timeout 60
+    ///   </code>
+    ///   <para>This will run a query in the book_data.classics table and will wait up to 60 seconds 
+    ///   for its completion.  When it finishes, it will print a number of BigQueryRow objects to 
+    ///   the terminal.</para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs)">
+    /// [BigQuery Jobs]
+    /// </para>
+    /// </summary>
+    [Cmdlet("Receive", "BqJob")]
+    public class ReceiveBqJob : BqCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// JobReference to get results from. Other types accepted are Job and JobsData.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 0, ValueFromPipeline = true)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(JobList.JobsData),
+            Property = nameof(JobList.JobsData.JobReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Apis.Bigquery.v2.Data.Job),
+            Property = nameof(Apis.Bigquery.v2.Data.Job.JobReference))]
+        [ValidateNotNull]
+        public JobReference InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Max time, in seconds, to wait for the job to complete before failing (Default: 10).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public int Timeout { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            Project = InputObject.ProjectId;
+            var options = new GetQueryResultsOptions();
+            options.Timeout = new TimeSpan(0, 0, (Timeout < 10) ? 10 : Timeout);
+            BigQueryResults result;
+            ulong rowsRead = 0;
+
+            do
+            {
+                options.StartIndex = rowsRead;
+                result = Client.GetQueryResults(InputObject, options);
+                foreach (BigQueryRow row in result.GetRows())
+                {
+                    WriteObject(row);
+                    rowsRead++;
+                }
+            } while (rowsRead < result.TotalRows);
```

Comment:
Do you need this outer loop? It looks to me like `result.GetRows()` will get them all.

> Returns a lazy enumeration of all the rows in the query (from any originally-specified page token onwards), making server calls as required. If the job has not completed already, it is polled until it has completed using pollSettings.

If you don't need the outerloop, you can just do `WriteObject(result.GetRows(), true)`.